### PR TITLE
Added permissions for FlatFile

### PR DESCRIPTION
**Pull request by davidsteinsland**
Ref issue #229.

Please tag after merge..? :hankey:

### CODE REVIEW

File: src/Bernard/Driver/FlatFileDriver.php [hunk 1]
```diff
@@ -18,13 +18,20 @@ class FlatFileDriver implements Driver
...
+    private $permissions;
+
+    /**
      * Constructor
      *
      * @param string $baseDirectory The base directory
+     * @param int    $permissions   Permissions to create the file with.
      */
-    public function __construct($baseDirectory)
+    public function __construct($baseDirectory, $permissions = 0)
```
Comment (henrikbjorn) [inline src/Bernard/Driver/FlatFileDriver.php]: Please have this be a suitable default of bits instead of `0`

Comment (davidsteinsland) [inline src/Bernard/Driver/FlatFileDriver.php]: I thought about this, and considered that users _may_ have set permissions via:
https://secure.php.net/umask

If I introduce a new default here, it _may_ break other applications. 

So, either we must set an acceptable default `0760` or something, or set permissions only when explicitly provided by the user.

Comment (henrikbjorn) [inline src/Bernard/Driver/FlatFileDriver.php]: I think we should set an acceptable default, also i think Doctrine does this for sqlite databases.

Comment (davidsteinsland) [inline src/Bernard/Driver/FlatFileDriver.php]: Set default permissions to `0740`.

**Pull request closed without merging**